lib/msf/core/post/windows/priv.rb

```diff
@@ -124,23 +124,19 @@ def is_system?
   # if running on a system that does not have UAC
   #
   def is_uac_enabled?
-    uac = false
-    winversion = session.sys.config.sysinfo['OS']
+    return false if is_system?
 
-    if winversion =~ /Windows (Vista|7|8|2008|2012|10)/
-      unless is_system?
-        begin
-          enable_lua = registry_getvaldata(
-              'HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System',
-              'EnableLUA'
-          )
-          uac = (enable_lua == 1)
-        rescue Rex::Post::Meterpreter::RequestError => e
-          print_error("Error Checking if UAC is Enabled: #{e.class} #{e}")
-        end
-      end
-    end
-    return uac
+    winversion = session.sys.config.sysinfo['OS']
+    return false unless winversion =~ /Windows (Vista|7|8|10|2008|2012|2016)/
+
+    enable_lua = registry_getvaldata(
```

Comment:
I don't think we even need to assign this variable anymore. We could do:
```ruby
1 == registry_getvaldata(...)
```

Reply:
Correct.

I thought that might be taking it a step too far. The variable deceleration is descriptive, while unnecessary. Additionally, the conditional should be wrapped in `( )` so it's obvious that it's conditional and not a variable assignment, which then looks a little silly:

```ruby
          (1 == registry_getvaldata(
              'HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System',
              'EnableLUA'
          ))
```

Reply:
Not that I care either way.

I'm a little surprised the existing code even works as intended. A quick look into the `registry_getvaldata` method in `lib/msf/core/post/windows/registry.rb` gives no indication of the expected datatype to be returned. I would have assumed string, rather than integer. This is important, because:

```
# irb
2.3.0 :001 > 1 == 1
 => true 
2.3.0 :002 > 1 == '1'
 => false 
2.3.0 :003 > 1 == "1"
 => false 
2.3.0 :004 > '1' == 1
 => false 
2.3.0 :005 > "1" == 1
 => false 
```

Reply:
@acammack-r7

Reply:
The registry has native integer types, so it doesn't surprise me that we return them. Since this isn't part of an assignment I don't thing parens are needed. If you want to make it an explicit `return 1 == ...` that would probably convey the meaning better.

Reply:
Updated to use `eql` instead.

>  Since this isn't part of an assignment I don't thing parens are needed.

I disagree. Braces are important to differentiate between comparison and assignment at a glance.

Forgoing braces and instead using a `return` still looks silly and effectively replaces a useless variable deceleration with a syntactically useless `return`.